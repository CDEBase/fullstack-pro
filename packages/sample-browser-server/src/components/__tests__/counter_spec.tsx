// // tslint:disable-next-line no-unused-variable
// import 'jest';
// import * as React from 'react'
// import * as TestUtils from 'react-addons-test-utils'

// import { createStore } from 'redux'

// import { Counter } from '../counter'
// import { reducers } from '../../reducers'

// describe('components/Counter', () => {

//   it('renders', () => {
//     const store = createStore(reducers)
//     const renderer = TestUtils.createRenderer()
//     expect(renderer.render(
//       <Counter label='a counter!' store={store} />
//     )).toMatchSnapshot()
//   })

//   describe('clicking "increment"', () => {
//     it('increments counter', () => {
//       const store = createStore(reducers)
//       const counter = TestUtils.renderIntoDocument(
//         <Counter label='a counter!' store={store} />
//       )
//       const [
//         increment,
//       ] = TestUtils.scryRenderedDOMComponentsWithTag(counter, 'button')
//       TestUtils.Simulate.click(increment)
//       TestUtils.Simulate.click(increment)
//       TestUtils.Simulate.click(increment)

//       const pre = TestUtils.findRenderedDOMComponentWithTag(counter, 'pre')
//       expect(JSON.parse(pre.textContent).counter.value).toEqual(3)
//     })
//   })
// })
