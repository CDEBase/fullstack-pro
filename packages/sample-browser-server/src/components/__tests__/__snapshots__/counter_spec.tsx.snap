exports[`components/Counter renders 1`] = `
<CounterComponent
  counter={
    Object {
      "value": 0,
    }
  }
  error=""
  increment={[Function]}
  isLoading={false}
  isSaving={false}
  label="a counter!"
  load={[Function]}
  save={[Function]}
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  } />
`;
